include/aspect/newton.h

```diff
@@ -105,6 +105,69 @@ namespace aspect
        */
       void set_newton_derivative_scaling_factor(const double newton_derivative_scaling_factor);
 
+      /**
+       * Get the stabilization type used in the preconditioner.
+       */
+      std::string get_preconditioner_stabilization() const;
+
+      /**
+       * Set the stabilization type used in the preconditioner.
+       */
+      void set_preconditioner_stabilization(const std::string preconditioner_stabilization);
+
+      /**
+       * Get the stabilization type used in the velocity block.
+       */
+      std::string get_velocity_block_stabilization() const;
+
+      /**
+       * Sets the stabilization type used in the velocity block.
+       */
+      void set_velocity_block_stabilization(const std::string velocity_block_stabilization);
+
+      /**
+       * Get whether to use the Newton failsafe
+       */
+      bool get_use_Newton_failsafe();
```

Comment:
can you add another sentence to the documentation that explains what the "failsafe" is?

Also end the sentence with a period.